code/modules/reagents/chemistry/reagents/drinks/alcohol_reagents.dm

```diff
@@ -589,39 +589,6 @@
 	taste_description = "oranges"
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
-/datum/reagent/consumable/ethanol/screwdrivercocktail/on_transfer(atom/atom, methods = TOUCH, trans_volume)
-	if(!(methods & INGEST))
-		return ..()
```

Comment:
This shouldn't be using `on_transfer` at all, as it's only called if methods is not zero or null. Furthermore, I'm certain the `INGEST` control statement won't pass for drinks because I'm pretty sure it's only for mobs. That's why it isn't working. 

That said, it should be done using `on_new()` instead, like here:
https://github.com/tgstation/tgstation/blob/58fdc807c6db231717e5cdf5dda92717db1000be/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm#L103-L106

Reply:
`on_new` doesn't seem very right either, wouldn't that mean it ONLY works if you mix the drink in the glass? Not if you pour from mixer -> glass?

Reply:
No, transferring a reagent from a holder to another generates a new instance of said reagent, so that won't be an issue.

Reply:
Yes, I realized what you said about why the old code didn't work when doing research for the pr. It _should_ be mentioned in the writeup in the respective issue but I may have forgotten to.

Regardless, the reason I decided to put it on the drinking glass rather than the reagent is that it only ever applies to the drinking glass anyway, and felt cleaner than making the reagent register a pile of signals to check whether or not it is still the master reagent when its contents get changed.

Now I'll look into it anyway, there's bound to be a way to do it cleaner for posterity use knowing this.

Reply:
Alright, so, while this seems to be doable relatively cleanly by using `on_new(...)` to register a proc for a `get_master_reagent()`-based toggle on
```dm
COMSIG_REAGENTS_ADD_REAGENT,
COMSIG_REAGENTS_NEW_REAGENT,
COMSIG_REAGENTS_REM_REAGENT
```
and resetting procs on
```dm
COMSIG_REAGENTS_DEL_REAGENT,
COMSIG_REAGENTS_CLEAR_REAGENTS
```
this approach proceeds to make it *also* work on shot glasses.

This is an issue of shot glasses being subtype to drinking glasses, meaning we can't check for the `/obj/item/reagent_containers/cup/glass/drinkingglass` type and expect it to *only* work on drinking glasses.
The implementation I used circumvented this by checking for the screwdriver style, which doesn't show up for shot glasses.

Now I get not wanting to hardcode it to the drinking glass, but currently I believe this is the _only_ reagent which modifies the tool behaviour of the container and it only works on the drinking glass, and using any purely signal on reagent method feels like it would not have most of the benefits anyway given it might lead to priority conflicts down the line, say:
- Add 5u hypothetical crowbar drink to 1u crowbar 5u screwdriver, crowbar sets tool behaviour, screwdriver resets tool behaviour.

Or if we added checks for current tool behaviour, say:
- Add 5u hypothetical screwdriver2 reagent to 1u screwdriver2 5u screwdriver, screwdriver2 sets tool behaviour, screwdriver sees it's still its tool type and resets tool behaviour.

So I _feel_ if we were to not do it on drinking glass, we'd need to use a different approach more parallel to the container styles component, which feels outside of the scope of this bugfix pr.

If you feel I'm overlooking something obvious, do feel free to say such. :P

Reply:
Good observation on the matter. I'd have suggested an approach like that. About the potential concurrency issue with hypothetical similar-functioning reagents, that's an (not) issue for another day, because this is the only one, and I doubt anyone is planning to add more soon.